### DISCUSSION TITLE
Suggestions for fitting Harmonic FC of large defective system

### DISCUSSION THREAD

**Discussion by nwinner** [ttadano/alamode#129] [Q&A]
I am a student trying to use Alamode to calculate the free energy of vacancies, first in the QHA, and ultimately with the SCPH approach. My example is a 108 atom monolayer supercell. In order to have matching k-point sampling between bulk and defective systems, I calculated harmonic force constants in the pristine, 108 atom supercell, getting a fitting error of ~6%. While not as low as some examples in the documentation, I consider this good-enough.

When I move to the defective supercell, I'm unable to get a good fit. 

I use a DFTSET of 108 randomly perturbed supercells (0.01Å distance), assuming worst case. The alamode run lead to a fitting error of ~48%. My first thought was I needed a larger dataset. So I increased the number of rattled supercells until my DFSET was 224 structures, but this lead to the exact same fitting error of 48%. So, there must be something else happening.

Would anyone be able to suggest some other strategies/things to try that could improve the fit for my FCs? I see @ttadano has been on at least one publication using alamode on defective systems, so I hope he can give some pointers if not too busy.

Thanks.

**Comment by ttadano** [comment]
The force components in the DFSET may include considerably large residual terms (the linear term of the potential energy) if the structure was not optimized after introducing a defect. 

If you didn't subtract the residual components, I would suggest trying it. 
https://alamode.readthedocs.io/en/latest/faq.html
